### PR TITLE
Significant progress

### PR DESCRIPTION
**Pull request by erikprice**

### CODE REVIEW

File: ComposableArchitecture/App/Sources/AppState.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    var search: Search
+}
+
+struct Search {
+    var searchQuery: String
+    var searchResults: [Word]
+}
+
+typealias EnglishWord = (String, Language)
```
Comment (caipre) [inline ComposableArchitecture/App/Sources/AppState.swift:17]: Just to recap the discussion we had, `EnglishWord` is a temporary value acting as an index key until we have a better model for the dictionary datasets we'll have. The other structure we discussed was to have, eg, `var todaysWord: Int` where the integer value is the primary key for the word irrespective of language. To keep things simple and human readable (where human is an English speaker), we'll just use the English word as the primary key for the time being.

File: ComposableArchitecture/App/Sources/AppState.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+typealias Word = String
+
+enum Language {
+    case en
+    case jp
+    case fr
+}
```
Comment (caipre) [inline ComposableArchitecture/App/Sources/AppState.swift:24]: If we want to use a third-party library, I found https://github.com/SwiftPackageRepository/ISO639.swift

File: ComposableArchitecture/App/Sources/AppState.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    var currentLanguage: Language
+
+    var search: Search
+}
+
+struct Search {
+    var searchQuery: String
+    var searchResults: [Word]
+}
```
Comment (caipre) [inline ComposableArchitecture/App/Sources/AppState.swift]:
```suggestion
struct Search {
    var query: String
    var results: [Word]
}
```

**Review by caipre — APPROVED** [review]

File: ComposableArchitecture/App/Sources/AppState.swift (hunk 3 above)
Comment (erikprice) [reply ComposableArchitecture/App/Sources/AppState.swift]: Agreed, that is an improvement.

File: ComposableArchitecture/App/Sources/AppState.swift (hunk 2 above)
Comment (erikprice) [reply ComposableArchitecture/App/Sources/AppState.swift:24]: I think issues are disabled on this repo, but should we move this to something like that so we don't lose track of it?

Comment (caipre) [reply ComposableArchitecture/App/Sources/AppState.swift:24]: Enabled issues! 👍

**Pull request merged**